java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -896,6 +896,10 @@ public long getVersion(String tag) {
     }
   }
 
+  public SqlQuery sql(String sql) {
```

Comment:
We could add some comments about this api is uses datafusion, and the default table name is `dataset`.

Reply:
IMO, we can tell users the default name in the `@param sql`'s comment. And tell users they can configure some options to build a final sql query for `@return`'s comment.